Mixpanel/Mixpanel.m

```diff
@@ -305,8 +310,16 @@ - (void)createAlias:(NSString *)alias forDistinctID:(NSString *)distinctID
         MPLogError(@"%@ create alias called with empty distinct id: %@", self, distinctID);
         return;
     }
-    [self track:@"$create_alias" properties:@{ @"distinct_id": distinctID, @"alias": alias }];
-    [self flush];
+    if (![alias isEqualToString:distinctID]) {
+        [self track:@"$create_alias" properties:@{ @"distinct_id": distinctID, @"alias": alias }];
+        [self flush];
+        dispatch_async(self.serialQueue, ^{
```

Comment:
ah i think you want this above flush and maybe above track. the reason being that flush has a blocking network request in it so your setting/archiving could wait a long time. it's possible above track is best because track potentially flushes inside of it.